src/write_gcm.c

```diff
@@ -3105,6 +3146,64 @@ static void wg_json_Metric(json_ctx_t *jc,
   wg_json_map_close(jc);
 }
 
+static int wg_default_metric_value_type_from_ds_type(int ds_type) {
+  int metric_value_type = METRIC_VALUE_TYPE_UNKNOWN;
+
+  switch(ds_type) {
+    case DS_TYPE_GAUGE:
+      metric_value_type = METRIC_VALUE_TYPE_DOUBLE;
+      break;
+    case DS_TYPE_DERIVE:
+    case DS_TYPE_COUNTER:
+      metric_value_type = METRIC_VALUE_TYPE_INT64;
+      break;
+  }
+
+  return metric_value_type;
+}
+
+static int wg_metric_value_type_from_wg_payload_t(
+    const wg_payload_t *element) {
+  int metric_value_type = METRIC_VALUE_TYPE_UNKNOWN;
+
+  for (int i = 0; i < element->key.num_metadata_entries; ++i) {
+    wg_metadata_entry_t *entry = &element->key.metadata_entries[i];
+    if (strcmp(entry->key, custom_metric_value_type_key) == 0) {
+      if (strcmp(entry->value.value_text, "INT64") == 0) {
+        metric_value_type = METRIC_VALUE_TYPE_INT64;
```

Comment:
If you return here, then you'll know that at the end of the outer loop the value is still unknown, making the code much simpler...